net/download/service/src/main/java/de/danoeh/antennapod/net/download/service/feed/FeedUpdateWorker.java

```diff
@@ -127,36 +130,56 @@ private Notification createNotification(@Nullable List<Feed> toUpdate) {
                 .build();
     }
 
+    private void updateNotification(List<Feed> toUpdate) {
+        if (ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.POST_NOTIFICATIONS)
+                == PackageManager.PERMISSION_GRANTED) {
+            notificationManager.notify(R.id.notification_updating_feeds, createNotification(toUpdate));
+        }
+    }
+
     @NonNull
     @Override
     public ListenableFuture getForegroundInfoAsync() {
         return Futures.immediateFuture(new ForegroundInfo(R.id.notification_updating_feeds, createNotification(null)));
     }
 
     private void refreshFeeds(List<Feed> toUpdate, boolean force) {
-        while (!toUpdate.isEmpty()) {
-            if (isStopped()) {
-                return;
-            }
-            if (ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.POST_NOTIFICATIONS)
-                    == PackageManager.PERMISSION_GRANTED) {
-                notificationManager.notify(R.id.notification_updating_feeds, createNotification(toUpdate));
-            }
-            Feed feed = toUpdate.get(0);
-            try {
-                if (feed.isLocalFeed()) {
-                    LocalFeedUpdater.updateFeed(feed, getApplicationContext(), null);
-                } else {
-                    refreshFeed(feed, force);
+        //Unchecked is used to prevent a race condition where a feed finishes downloading before
+        // we're done submitting all of the jobs
+        List<Feed> unchecked = new ArrayList<Feed>(toUpdate);
+        updateNotification(unchecked);
+        ExecutorService executor = Executors.newFixedThreadPool(5);
```

Comment:
5 sounds a bit arbitrary. I think the default was 4 when we had parallel refresh before (which got removed after a rewrite of the download code)